src/metabase/util/i18n/impl.clj

```diff
@@ -203,31 +203,25 @@
              (log/errorf e "Invalid format string %s" (pr-str format-string))
              format-string)))))))
 
-;; We can't fetch the system locale until the application DB has been initiailized. Once that's done, we don't need to
-;; do the check anymore -- swapping out the getter fn with the simpler one speeds things up substantially
-(def ^:private site-locale-from-setting-fn
-  (atom
-   (fn []
-     (when-let [db-is-set-up? (resolve 'metabase.db/db-is-set-up?)]
-       (when (and (bound? db-is-set-up?)
-                  (db-is-set-up?))
-         (when-let [get-value-of-type (resolve 'metabase.models.setting/get-value-of-type)]
-           (when (bound? get-value-of-type)
-             (let [f (fn [] (get-value-of-type :string :site-locale))]
-               (reset! site-locale-from-setting-fn f)
-               (f)))))))))
```

Comment:
Here's another dumb optimization that saved about two tenths of a microsecond, it's simply not worth turning our code into a hairball to save a few fractions of a millionth of a second.

```
old impl

Evaluation count : 389046 in 6 samples of 64841 calls.
             Execution time mean : 1.545614 µs
    Execution time std-deviation : 6.218459 ns
   Execution time lower quantile : 1.539294 µs ( 2.5%)
   Execution time upper quantile : 1.552483 µs (97.5%)
                   Overhead used : 1.566506 ns

new impl

Evaluation count : 351720 in 6 samples of 58620 calls.
             Execution time mean : 1.741505 µs
    Execution time std-deviation : 6.093460 ns
   Execution time lower quantile : 1.735728 µs ( 2.5%)
   Execution time upper quantile : 1.750150 µs (97.5%)
                   Overhead used : 1.566506 ns
```

Also `metabase.db.setting/db-or-cache-value` already checks `db-is-set-up?`, so this check was a duplicate.